src/lib/DatePicker.svelte

```diff
@@ -7,19 +7,29 @@
 
   const dispatch = createEventDispatcher<{ select: undefined }>()
 
+  /** Default Date to use */
+  const defaultDate = new Date()
+
   /** Date value */
-  export let value = new Date()
+  export let value: Date | null = null
   function setValue(d: Date) {
-    if (d.getTime() !== value.getTime()) value = d
+    if (d.getTime() !== value?.getTime()) value = d
+  }
+
+  let tmpPickerDate = value ?? defaultDate
+  $: tmpPickerDate = value ?? defaultDate
+
+  function setPickerDate(d: Date) {
+    if (d.getTime() !== tmpPickerDate.getTime()) tmpPickerDate = d
   }
-  function updateValue(updater: (date: Date) => Date) {
-    let d = updater(new Date(value.getTime()))
-    setValue(d)
+  function updatePickerDate(updater: (date: Date) => Date) {
+    let d = updater(new Date(tmpPickerDate.getTime()))
+    setPickerDate(d)
   }
```

Comment:
The logic for `setPickerDate` / `updatePickerDate` doesn't seem right. Picking a dayOfMonth should always update the `value`. Picking a new month or year should update the `value` only if `value` is not null.

Reply:
This is tricksy - if you change the month before selecting a day then we don't know which day of the month to use and probably should not assign a value (feels like a side effect to me). If you've selected a day and then change the month (or year) then it makes sense to adjust the date with the new month (or year). we could make that adjustment if the value has already been assigned but i think it should not be populated if they've only selected month or year and have not yet selected a date.  need to test this in the browser again....

Reply:
It wouldn't be a problem because you'd only assign to `value` if it's not null. If `value === null`, then we would assign to `shownDate` (or `tmpPickerDate`) instead so that the popup is updated, but the `value` stays unchanged

Reply:
ok -- after playing with it I totally agree with your assessment. I updated the PR to adjust the value if the value is defined when the month or year is changed.

Reply:
Seems to work now, but I think it's better if `updatePickerDate` takes care of updating `value`. I went and made that happen in the [nullable branch](https://github.com/probablykasper/date-picker-svelte/tree/nullable), feel free to merge it in (also renamed `tmpPickerDate` to `shownDate`).

Reply:
@probablykasper -- I merged in your nullable branch. I noticed one issue:

If you type in the date -  you can set a date outside the min/max.  need to take a look at that one - doesn't happen with the master branch.

Reply:
Yes I noticed as well, my latest comment is about that one. I'll add it to a suggestion so it's easy to merge in

Reply:
nvm can't add it as a suggestion, but the solution is there anyways